map/bookmark_manager.cpp

```diff
@@ -879,16 +873,13 @@ void BookmarkManager::SetElevationMyPositionChangedCallback(
   m_elevationMyPositionChanged = cb;
 }
 
-void BookmarkManager::SetElevationActivePoint(kml::TrackId const & trackId, double targetDistance)
+void BookmarkManager::SetElevationActivePoint(kml::TrackId const & trackId, m2::PointD pt, double targetDistance)
 {
   CHECK_THREAD_CHECKER(m_threadChecker, ());
 
   auto const track = GetTrack(trackId);
   CHECK(track != nullptr, ());
 
-  m2::PointD pt;
-  VERIFY(track->GetPoint(targetDistance, pt), (trackId, targetDistance));
```

Comment:
The pr and dist are used because the TrackSelectionInfo struct should have this data:
https://github.com/organicmaps/organicmaps/blob/dfa2dd24d9816d74a092eb253d7a422b3ef3b43f/map/track.hpp#L34-L49

The old logic was to calculate the point coordinates based on the distance on every update. It is quite expensive and complex logic because the mark's coordinates calculation may be called frequently when the user drags the chart view:
https://github.com/organicmaps/organicmaps/blob/dfa2dd24d9816d74a092eb253d7a422b3ef3b43f/map/track.cpp#L13-L47

It is much easier and faster to use the cached points because we already have them in the multigomertry.